pkg/crd/osm/v1alpha1/operatingsystemprofile_types.go

```diff
@@ -51,8 +51,15 @@ type OperatingSystemProfileSpec struct {
 	Version string `json:"version"`
 	// SupportedCloudProviders represent the cloud providers that support the given operating system version
 	SupportedCloudProviders []CloudProviderSpec `json:"supportedCloudProviders"`
+	// Bootstrap config is used for initial configuration of machine and to fetch the kubernetes secret that contains the provisioning config.
+	BootstrapConfig OSPConfig `json:"bootstrapConfig"`
+	// Provisioning Config is used for provisioning the worker node.
+	ProvisioningConfig OSPConfig `json:"provisioningConfig"`
+}
+
+type OSPConfig struct {
 	// SupportedContainerRuntimes represents the container runtimes supported by the given OS
-	SupportedContainerRuntimes []ContainerRuntimeSpec `json:"supportedContainerRuntimes"`
+	SupportedContainerRuntimes []ContainerRuntimeSpec `json:"supportedContainerRuntimes,omitempty"`
```

Comment:
It's intentional, this is not a `required` field anymore. This is part of the [OSPConfig](https://github.com/kubermatic/operating-system-manager/blob/7651eea4a849ddaeacfd5ad20af85b43e3fcd22a/pkg/crd/osm/v1alpha1/operatingsystemprofile_types.go#L60) struct which is used in bootstrap and provisioning config. Bootstrapping configuration usually won't have any container runtime specific configurations.

Also, even in general it makes sense to mark this as an optional field since it's up to the users to decide if they want to add container runtime specific configurations in their custom OSPs.